src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80CompoundDirectory.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.KNN80Codec;
+
+import lombok.Getter;
+import org.apache.lucene.codecs.CompoundDirectory;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.IOContext;
+import org.apache.lucene.store.IndexInput;
+
+import java.io.IOException;
+import java.util.Set;
+
+public class KNN80CompoundDirectory extends CompoundDirectory {
```

Comment:
in the refresh stack
`IndexShard.openEngineAndRecoverFromTranslog` -> `InternalEngine.refresh`  -> `StandardDirectoryReader.open` -> `SegmentDocValues.newDocValuesProducer` -> `KNN80DocValuesFormat.fieldsProducer` it would throw Exception

> java.lang.ClassCastException: class org.apache.lucene.codecs.lucene90.Lucene90CompoundReader cannot be cast to class org.apache.lucene.store.FSDirectory (org.apache.lucene.codecs.lucene90.Lucene90CompoundReader and org.apache.lucene.store.FSDirectory are in unnamed module of loader 'app')

i think it is because `Lucene90CompoundReader` is an final class, we can not cast it directly across module. we need an wrap for `Lucene#Codec#CompoundDirectory` which reader is already be an wrap directory